api/v1beta1/machine_types.go

```diff
@@ -143,6 +168,13 @@ type MachineSpec struct {
 	NodeDeletionTimeout *metav1.Duration `json:"nodeDeletionTimeout,omitempty"`
 }
 
+// MachineReadinessGate contains the type of a Machine condition to be used as a readiness gate.
+type MachineReadinessGate struct {
+	// conditionType refers to a condition with matching type in the Machine's condition list.
```

Comment:
should we doc that the expectation is for it to be true?

Reply:
added